components/dashboard/src/settings/Integrations.tsx

```diff
@@ -199,77 +179,435 @@ function GitProviders() {
+            <a href="https://www.gitpod.io/docs/gitlab-integration/#oauth-application" target="_blank" rel="noopener" className="text-gray-400 underline underline-thickness-thin underline-offset-small hover:text-gray-600">Learn more</a>.
+        </span>);
+    }
+
+    const copyRedirectUrl = () => {
+        const el = document.createElement("textarea");
+        el.value = redirectURL;
+        document.body.appendChild(el);
+        el.select();
+        try {
+            document.execCommand("copy");
+        } finally {
+            document.body.removeChild(el);
+        }
+    };
+
+    return (<Modal visible={!!props} onClose={close}>
+        <h3 className="pb-2">{props.mode === "new" ? "New Git Integration" : "Git Integration"}</h3>
+        <div className="space-y-4 border-t border-b border-gray-200 mt-2 -mx-6 px-6 py-4">
+            {props.mode === "edit" && props.provider.status === "pending" && (
+                <div className="flex rounded-md bg-gitpod-kumquat-light p-3">
+                    <span className="text-red-600">You need to actiate this integration.</span>
+                </div>
+            )}
+            <div className="flex flex-col">
+                <span className="text-gray-500">Configure a git integration with a GitLab or GitHub self-hosted instance.</span>
+            </div>
+            {props.mode === "new" && (
+                <div className="flex flex-col space-y-2">
+                    <label htmlFor="type" className="font-medium">Provider Type</label>
+                    <select name="type" value={type} disabled={props.mode !== "new"} className="rounded-md w-full"
+                        onChange={(e) => setType(e.target.value)}>
+                        <option value="GitHub">GitHub</option>
+                        <option value="GitLab">GitLab</option>
+                    </select>
+                </div>
+            )}
+            <div className="flex flex-col space-y-2">
+                <label htmlFor="hostName" className="font-medium">Provider Host Name</label>
+                <input name="hostName" disabled={props.mode === "edit"} type="text" value={host} className="rounded-md w-full"
+                    onChange={(e) => updateHostValue(e.target.value)} />
+            </div>
+            <div className="flex flex-col space-y-2">
+                <label htmlFor="redirectURL" className="font-medium">Redirect URL</label>
+                <div className="w-full relative">
+                    <input name="redirectURL" disabled={true} readOnly={true} type="text" value={redirectURL} className="rounded-md w-full" />
+                    <div className="cursor-pointer" onClick={() => copyRedirectUrl()}>
+                        <img src={images.copy} className="absolute top-1/3 right-3" />
```

Comment:
**issue**: This is missing some tooltip which could cause some usability issues. Feel free to leave this as is as we can tackle this in the a later iteration in https://github.com/gitpod-io/gitpod/issues/3592.

Reply:
adding simple `title` for now.